src/main/java/com/minecolonies/core/entity/pathfinding/navigation/PathfindingAIHelper.java

```diff
@@ -23,19 +23,28 @@ public static boolean walkCloseToXNearY(
     {
         final MinecoloniesAdvancedPathNavigate nav = ((MinecoloniesAdvancedPathNavigate) entity.getNavigation());
 
-        if (nav.isDone() || (nav.getPathResult() != null
-                               && !(nav.getPathResult().getJob() instanceof PathJobMoveCloseToXNearY job
-                                      && job.nearbyPosition.equals(nearbyPosition)
-                                      && job.desiredPosition.equals(desiredPosition)
-                                      && job.distToDesired == distToDesired)))
+        // Three cases
+        // 1. Navigation Finished
+        // 2. Navigation is progressing towards a previous task
+        // 3. Navigation did not try once
+        boolean isOnRightTask = (nav.getPathResult() != null
+            && nav.getPathResult().getJob() instanceof PathJobMoveCloseToXNearY job
+            && job.nearbyPosition.equals(nearbyPosition)
+            && job.desiredPosition.equals(desiredPosition));
+
+        if (nav.isDone() || !isOnRightTask)
         {
-            // Check distance once navigation is done, to let the entity walk
-            if (BlockPosUtil.dist(entity.blockPosition(), desiredPosition) < distToDesired)
+            if (isOnRightTask)
             {
-                return false;
+                // Check distance once navigation is done, to let the entity walk
+                if (BlockPosUtil.dist(entity.blockPosition(), desiredPosition) < distToDesired)
+                {
+                    nav.stop();
+                    return false;
+                }
             }
 
-            PathJobMoveCloseToXNearY pathJob = new PathJobMoveCloseToXNearY(entity.level, desiredPosition, nearbyPosition, distToDesired, entity);
+            PathJobMoveCloseToXNearY pathJob = new PathJobMoveCloseToXNearY(entity.level, desiredPosition, nearbyPosition, 1, entity);
```

Comment:
Intentionally only 1 dist?

Reply:
ye to have the entity move as close as possible